modules/network/src/integrationTest/java/org/apache/ignite/internal/network/scalecube/ArbitraryProductVersionSource.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.scalecube;
+
+import org.apache.ignite.internal.properties.IgniteProductVersion;
+import org.apache.ignite.internal.version.ProductVersionSource;
+
+/**
+ * {@link ProductVersionSource} that allows arbitrary product name and version to be specified.
+ */
+class ArbitraryProductVersionSource implements ProductVersionSource {
```

Comment:
As far as I understand, this class is used in tests, let's add the prefix `Test` to it?

Reply:
It is just an alternative implementation of the interface. It's only used for tests, so it's placed under `integrationTest` tree, which seems enough. I think it's more useful if a type name defines just main feature of the type (in this case, it's that arbitrary product name and version can be specified) and not some secondary features (like whether it's only used for tests)

Reply:
Okay, it's up to you.